### DISCUSSION TITLE
Clarification about cluster changes when in minority

### DISCUSSION THREAD

**Discussion by visciang** [rabbitmq/ra#389] [General]
Hello!

I've been playing with a ra cluster of 3 nodes: A, B, C.
Cluster successfully established, with leader B.
I've shutdown the two followers A and C leaving the leader B alone.

At this point I've tried to change the cluster topology adding a new node D.
The node has been accepted by B, and D has a "follower" role.

Is this what we should expect, being in minority?

Then I've tried to add another node E.
In this case the operation is instead rejected with cause cluster_change_not_permitted.

Why?

Attaching here an example to reproduce the above steps (this has been tested with ra v2.6.3).

Thank you!
Giovanni

[test.txt](https://github.com/rabbitmq/ra/files/12241642/test.txt)

**Comment by lessness** [comment]
Any news on this? Looks like a bug.

   Federico

**Comment by kjnilsson** [comment]
If you have a leader you can always make one member change. The member changes take effect when adding the cluster change command to the log _not_ on consensus as perhaps is what you expect. See the raft paper/thesis for more details. 

Closing as expected behaviour.
